src/core/expr/fexpr_nth.cc

```diff
@@ -0,0 +1,167 @@
+      out += ", skipna=";
+      out += skipna_;
+      out += ')';
+      return out;
+    }
+  
+
+    Workframe evaluate_n(EvalContext &ctx) const override {
+      Workframe inputs = arg_->evaluate_n(ctx);
+      Workframe outputs(ctx);
+      Groupby gby = ctx.get_groupby();
+      colvec col_s;
+      col_s.reserve(inputs.ncols());
+      if (!gby) gby = Groupby::single_group(ctx.nrows()); 
+      for (size_t i = 0; i < inputs.ncols(); ++i) {
+        auto coli = evaluate2(inputs.retrieve_column(i));
+        col_s.push_back(coli);
+      }
+      FExpr_RowAll::apply_function(std::move(col_s), inputs.nrows(), inputs.ncols());
+
+      // for (size_t i = 0; i < inputs.ncols(); ++i) {
+      //   auto coli = inputs.retrieve_column(i);
+      //   outputs.add_column(
+      //     evaluate1(std::move(coli), gby, n_),
+      //     inputs.retrieve_name(i),
+      //     Grouping::GtoONE
+      //   );
+      // }
+      return outputs;
+    }
+
+
+    Column evaluate1(Column&& col, const Groupby& gby, const int32_t n) const {
+      SType stype = col.stype();
+      switch (stype) {
+        case SType::VOID:    return Column(new ConstNa_ColumnImpl(gby.size()));
+        case SType::BOOL:
+        case SType::INT8:    return make<int8_t>(std::move(col), gby, n);
+        case SType::INT16:   return make<int16_t>(std::move(col), gby, n);
+        case SType::DATE32:
+        case SType::INT32:   return make<int32_t>(std::move(col), gby, n);
+        case SType::TIME64:
+        case SType::INT64:   return make<int64_t>(std::move(col), gby, n);
+        case SType::FLOAT32: return make<float>(std::move(col), gby, n);
+        case SType::FLOAT64: return make<double>(std::move(col), gby, n);
+        case SType::STR32:   return make<CString>(std::move(col), gby, n);
+        case SType::STR64:   return make<CString>(std::move(col), gby, n);
+        default:
+          throw TypeError()
+            << "Invalid column of type `" << stype << "` in " << repr();
+      }
+    }
+
+    Column evaluate2(Column&& col) const {
+      switch (col.stype()) {
+        case SType::VOID:    return Const_ColumnImpl::make_bool_column(col.nrows(), true);
+        case SType::BOOL:
+        case SType::INT8:    return Column(new Isna_ColumnImpl<int8_t>(std::move(col)));
+        case SType::INT16:   return Column(new Isna_ColumnImpl<int16_t>(std::move(col)));
+        case SType::DATE32:
+        case SType::INT32:   return Column(new Isna_ColumnImpl<int32_t>(std::move(col)));
+        case SType::TIME64:
+        case SType::INT64:   return Column(new Isna_ColumnImpl<int64_t>(std::move(col)));
+        case SType::FLOAT32: return Column(new Isna_ColumnImpl<float>(std::move(col)));
+        case SType::FLOAT64: return Column(new Isna_ColumnImpl<double>(std::move(col)));
+        case SType::STR32:
+        case SType::STR64:  return Column(new Isna_ColumnImpl<CString>(std::move(col)));
+        default: throw RuntimeError();
+      }
+    }
```

Comment:
So what is the purpose of the `evaluate2` method and why `evaluate1` is commented out?

Reply:
kindly ignore that - I was experimenting with the `FExpr_RowAll` function, to see how it works